### PR TITLE
Fix: Do not crash when event processors throw a lower level Throwable class

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: Do not crash when event processors throw a lower level Throwable class

## :bulb: Motivation and Context
https://github.com/getsentry/sentry-unity/issues/398

## :green_heart: How did you test it?
running it on newer Unity versions

## :pencil: Checklist

- [X] I reviewed the submitted code
- [X] I added tests to verify the changes
- [ ] I updated the docs if needed
- [X] No breaking changes

### CODE REVIEW

**Review by bitsandfoxes — APPROVED** [review]

**Review by romtsn — APPROVED** [review]
LGTM, but I have no idea what I'm doing anyway :D

File: sentry/src/main/java/io/sentry/SentryClient.java [hunk 1]
```diff
@@ -235,7 +235,7 @@ private SentryTransaction processTransaction(
     for (final EventProcessor processor : eventProcessors) {
       try {
         transaction = processor.process(transaction, hint);
-      } catch (Exception e) {
```
Comment (bruno-garcia) [inline sentry/src/main/java/io/sentry/SentryClient.java:238]: Do we have other places in the codebase we should make this change?

File: sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java [hunk 2]
```diff
@@ -32,6 +33,22 @@
         Objects.requireNonNull(activityFramesTracker, "ActivityFramesTracker is required");
   }
 
+  /**
+   * Returns the event itself
+   *
+   * @param event the SentryEvent the SentryEvent
+   * @param hint the Hint the Hint
```
Comment (lucas-zimerman) [inline sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java:40]: confusing 🤪

File: sentry/src/main/java/io/sentry/SentryClient.java (hunk 1 above)
Comment (marandaneto) [reply sentry/src/main/java/io/sentry/SentryClient.java:238]: maybe yes...

File: sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java (hunk 2 above)
Comment (marandaneto) [reply sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java:40]: indeed, it was auto-generated, didn't realize the confusion :(

**Pull request merged**